go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -634,14 +638,24 @@ func (qre *QueryExecutor) execDMLPKRows(conn *TxConnection, query *sqlparser.Par
 }
 
 func (qre *QueryExecutor) execSet() (*sqltypes.Result, error) {
-	conn, err := qre.getConn(qre.tsv.qe.conns)
+	pool := qre.getAppConnPool()
+	conn, err := qre.getConn(pool)
 	if err != nil {
 		return nil, err
 	}
 	defer conn.Recycle()
 	return qre.dbConnFetch(conn, qre.plan.FullQuery, qre.bindVars, nil, false)
 }
 
```

Comment:
just a style, i don't see it necessary to use `return param` assignment in this simple control flow, could just return from either branch